src/lib.rs

```diff
@@ -1407,6 +1411,7 @@ macro_rules! location {
 #[cfg(any(feature = "std", test))]
 pub struct Whatever {
     #[snafu(source(from(Box<dyn std::error::Error>, Some)))]
+    #[snafu(provide(false))]
```

Comment:
Great observation! `snafu(source)` will control `Error::source`, so that will continue to work. The rest is more complicated...

Because we don't know a useful concrete type of the underlying source, we couldn't implement something like `request_ref::<TheInnerError>()`. We could implement `request_ref::<dyn Error>`. More importantly, we don't chain off to the provided values of the underlying error, which is something I should fix.

That will mean that I'll be adding a fourth option to the attribute, something like:

```rust
#[snafu(provide(opt, ref, chain, dyn std::error::Error => source.as_deref()))]
```

- `opt` —> it's optional
- `ref` —> it's a reference
- `chain` —> if this value is present, add all of its provided data too.

Thoughts on the name `chain`? I also was thinking maybe `delegate`, but am open to any suggestions.

Reply:
This should now be working (still need to update the documentation to reflect it)

Reply:
And minimal docs / examples added.